src/Pillar-ExporterCore/PRTarget.class.st

```diff
@@ -8,15 +8,27 @@ Class {
 	#category : #'Pillar-ExporterCore-Base'
 }
 
+{ #category : #'as yet unclassified' }
+PRTarget class >> associateTarget: aFormat [
+	^ self allSubclasses
+		select: [ :each | (each isAbstract not) and: each builderName = aFormat asString ]
+]
+
 { #category : #clap }
 PRTarget class >> buildClapCommand [
 	
+	^ self buildClapCommandWithId: self clapId
+]
+
+{ #category : #clap }
+PRTarget class >> buildClapCommandWithId: anId [
+	
 	| target project |
 	target := self new.
 	project := self newProject.
  
-	^ (ClapCommand id: self clapId)
-		description: 'Build your Pillar documents in ', self clapId ,' format';
+	^ (ClapCommand id: anId asSymbol)
+		description: 'Build your Pillar documents in ', anId asString ,' format';
```

Comment:
This will be strange for the `#build` case.
It will generate a description `Build your Pillar documents in build format`.
Which is not correct.

We should somehow find a way to override the description  for the default case.

Reply:
Solved! I replace anId by self clapId, which is the name of export format.